src/main/groovy/com/twcable/gradle/sling/osgi/SlingOsgiBundle.groovy

```diff
@@ -424,24 +424,30 @@ class SlingOsgiBundle {
         def resolved = statuses[3] as Integer
         def installed = statuses[4] as Integer
 
+        if (resolved == 0 && installed == 0) {
+            log.debug "There are no bundles in the \"resolved\" or \"installed\" state"
+            return true
+        }
+
         List<Map> data = json.data as List
 
-        def inactiveBundles = data.findAll { bundle ->
-            bundle.state == INSTALLED.stateString ||
-                bundle.state == RESOLVED.stateString
-        }.collect { it.symbolicName } as List<String>
+        def inactiveBundles = inactiveBudles(data).collect { it.symbolicName } as List<String>
 
         if (log.infoEnabled) inactiveBundles.each { log.info "Inactive bundle: ${it}" }
 
-        if (resolved == 0 && installed == 0) {
+        if (inactiveBundles.isEmpty() || !inactiveBundles.any { it.contains(groupProperty) })
             return true
-        }
-        else if (!inactiveBundles.isEmpty() && inactiveBundles.every { it.contains(groupProperty) }) {
-            return true
-        }
-        else {
+        else
             return false
-        }
+    }
+
+
+    private static Collection<Map> inactiveBudles(Collection<Map> knownBundles) {
+        return knownBundles.findAll { bundle ->
+            bundle.state == INSTALLED.stateString ||
+                bundle.state == RESOLVED.stateString ||
+                bundle.state == MISSING.stateString
+        } as Collection<Map>
```

Comment:
The refactor from line # 429 above didn't have `bundle.state == MISSING.stateString` in the OR, but now it does ( as it uses this method and so does line # 460 ) . This may be intended, but just pointing it out.